bftcosi/bftcosi.go

```diff
@@ -1,798 +1,239 @@
-	signRefusal bool
-	// allowedExceptions for how much exception is allowed. If more than allowedExceptions number
-	// of conodes refuse to sign, no signature will be created.
-	allowedExceptions int
-	// our index in the Roster list
-	index int
-
-	// onet-channels used to communicate the protocol
-	// channel for announcement
-	announceChan chan announceChan
-	// channel for commitment
-	commitChan chan commitChan
-	// Two channels for the challenge through the 2 rounds: difference is that
-	// during the commit round, we need the previous signature of the "prepare"
-	// round.
-	// channel for challenge during the prepare phase
-	challengePrepareChan chan challengePrepareChan
-	// channel for challenge during the commit phase
-	challengeCommitChan chan challengeCommitChan
-	// channel for response
-	responseChan chan responseChan
-
-	// Internal communication channels
-	// channel used to wait for the verification of the block
-	verifyChan chan bool
-
-	// handler-functions
-	// onDone is the callback that will be called at the end of the
-	// protocol when all nodes have finished. Either at the end of the response
-	// phase of the commit round or at the end of a view change.
-	onDone func()
-	// onSignatureDone is the callback that will be called when a signature has
-	// been generated ( at the end of the response phase of the commit round)
-	onSignatureDone func(*BFTSignature)
-	// VerificationFunction will be called
-	// during the (start/handle) challenge prepare phase of the protocol
-	VerificationFunction VerificationFunction
-	// closing is true if the node is being shut down
-	closing bool
-	// mutex for closing down properly
-	closingMutex sync.Mutex
-}
-
-// collectStructs holds the variables that are used during the protocol to hold
-// messages
-type collectStructs struct {
-	// prepare-round cosi
-	prepare *crypto.CoSi
-	// commit-round cosi
-	commit *crypto.CoSi
-
-	// prepareSignature is the signature generated during the prepare phase
-	// This signature is adapted according to the exceptions that occured during
-	// the prepare phase.
-	prepareSignature []byte
-
-	// mutex for all temporary structures
-	tmpMutex sync.Mutex
-	// exceptions given during the rounds that is used in the signature
-	tempExceptions []Exception
-	// temporary buffer of "prepare" commitments
-	tempPrepareCommit []kyber.Point
-	// temporary buffer of "commit" commitments
-	tempCommitCommit []kyber.Point
-	// temporary buffer of "prepare" responses
-	tempPrepareResponse []kyber.Scalar
-	// temporary buffer of the public keys for nodes that responded
-	tempPrepareResponsePublics []kyber.Point
-	// temporary buffer of "commit" responses
-	tempCommitResponse []kyber.Scalar
+	// prepCosiProtoName is the cosi protocol name for the prepare phase
+	prepCosiProtoName string
+	// commitCosiProtoName is the cosi protocol name for the commit phase
+	commitCosiProtoName string
+	// prepSigChan is the channel for reading the prepare phase signature
```

Comment:
what is the []byte that goes through this channel? I understand it as a signal only, in which case it should be a struct{}.

Reply:
It's the signature from the prepare phase of bftcosi. It is used for verification in Dispatch.